### ISSUE TITLE
Possible bug

### ISSUE THREAD

**Issue by wbrenton** [bwfbowen/muax#2]
Did you intentionally not include the loss associated with 'initial inference' (prior to unrolling for k steps)? Possible that I am misunderstanding paper.

**Comment by bwfbowen** [comment]
Hi, thank you for reporting this issue. Could you provide more information about "loss associated with initial inference"?
Currently, the implementation follows the paper [Mastering Atari, Go, chess and shogi by planning with a learned model](https://www.nature.com/articles/s41586-020-03051-4). In the `loss/default_loss_fn`, the loss is calculated as follows.
Here each time the loss function will first encode the raw observation, and iteratively loop through the hidden states and calculate loss with actual observation. The `batch.Rn` is the bootstrapping value, where the value is calculated by sequence of rewards and value of the root node. 
I think the initial inference uses representation to encode the raw observation and prediction to provide `prior_logits` and `value`, so I think `default_loss_fn` includes this part. What do you think? Please share your thoughts. 

```Python
@partial(jax.jit, static_argnums=(0, ))
def default_loss_fn(muzero_instance, params, batch):
    
    # Use muzero_instance to access required methods and attributes
    loss = 0
    c = 1e-4
    B, L, _ = batch.a.shape
    batch.r = scalar_to_support(batch.r, muzero_instance._support_size).reshape(B, L, -1)
    batch.Rn = scalar_to_support(batch.Rn, muzero_instance._support_size).reshape(B, L, -1)
    s = muzero_instance._repr_apply(params['representation'], batch.obs[:, 0, :])
    
    def body_func(i, loss_s):
      loss, s = loss_s
      v, logits = muzero_instance._pred_apply(params['prediction'], s)
      # Appendix G, scale the gradient at the start of the dynamics function by 1/2 
      s = scale_gradient(s, 0.5)
      r, s = muzero_instance._dy_apply(params['dynamic'], s, batch.a[:, i, :].flatten())
      # losses: reward
      loss_r = jnp.mean(
        optax.softmax_cross_entropy(r, 
        jax.lax.stop_gradient(batch.r[:, i, :])
        ))
      # losses: value
      loss_v = jnp.mean(
        optax.softmax_cross_entropy(v, 
        jax.lax.stop_gradient(batch.Rn[:, i, :])
        ))
      # losses: action weights
      loss_pi = jnp.mean(
        optax.softmax_cross_entropy(logits, 
        jax.lax.stop_gradient(batch.pi[:, i, :])
        ))

      loss += loss_r + loss_v + loss_pi 
      loss_s = (loss, s)
      return loss_s 
    loss, _ = jax.lax.fori_loop(1, L, body_func, (loss, s))
    # Appendix G Training: "irrespective of how many steps we unroll for"
    loss /= L 

    # L2 regulariser
    l2_regulariser = 0.5 * sum(
      jnp.sum(jnp.square(p)) for p in jax.tree_util.tree_leaves(params))
    loss += c * jnp.sum(l2_regulariser)
    # print(f'loss2: {loss}')
    return loss 
```

**Comment by wbrenton** [comment]
Want to connect via email or discord? Either way you can reach me at the email in my bio.

**Comment by bwfbowen** [comment]
How about email? I have also included my email in bio(just now)
